vm/actor/src/builtin/power/policy.rs

```diff
@@ -1,107 +1,66 @@
-pub const CONSENSUS_MINER_MIN_MINERS: usize = 3;
-
-/// Maximum age of a block header used as proof of a consensus fault to appear in the chain.
-pub const CONSENSUS_FAULT_REPORTING_WINDOW: ChainEpoch = 2880; // 1 day @ 30 second epochs.
-
 lazy_static! {
-    /// Multiplier on sector pledge requirement.
-    pub static ref PLEDGE_FACTOR: BigUint = BigUint::from(3u8); // PARAM_FINISH
-
-    /// Total expected block reward per epoch (per-winner reward * expected winners), as input to pledge requirement.
-    pub static ref EPOCH_TOTAL_EXPECTED_REWARD: BigUint = reward::BLOCK_REWARD_TARGET.clone() * 5u8; // PARAM_FINISH
-
     /// Minimum power of an individual miner to meet the threshold for leader election.
     pub static ref CONSENSUS_MINER_MIN_POWER: StoragePower = StoragePower::from_i32(2 << 30).unwrap(); // placeholder
 
+    pub static ref BASE_MULTIPLIER: BigInt = BigInt::from(10); // PARAM_FINISH
+    pub static ref DEAL_WEIGHT_MULTIPLIER: BigInt = BigInt::from(11); // PARAM_FINISH
+    pub static ref VERIFIED_DEAL_WEIGHT_MULITPLIER: BigInt = BigInt::from(100); // PARAM_FINISH
+    pub static ref SECTOR_QUALITY_PRECISION: BigInt =  BigInt::from(20); // PARAM_FINISH
 }
 
-/// Penalty to pledge collateral for the termination of an individual sector.
-pub(super) fn pledge_penalty_for_sector_termination(
-    _pledge: &TokenAmount,
-    _term_type: SectorTermination,
-) -> TokenAmount {
-    // PARAM_FINISH
-    TokenAmount::from(0u8)
-}
-
-// Penalty to pledge collateral for repeated failure to prove storage.
-pub(super) fn pledge_penalty_for_windowed_post_failure(
-    _pledge: &TokenAmount,
-    _term_type: SectorTermination,
-) -> TokenAmount {
-    // PARAM_FINISH
-    TokenAmount::from(0u8)
-}
-
-/// Penalty to pledge collateral for a consensus fault.
-pub(super) fn pledge_penalty_for_consensus_fault(
-    pledge: TokenAmount,
-    fault_type: ConsensusFaultType,
-) -> TokenAmount {
-    // PARAM_FINISH: always penalise the entire pledge.
-    match fault_type {
-        ConsensusFaultType::DoubleForkMining => pledge,
-        ConsensusFaultType::ParentGrinding => pledge,
-        ConsensusFaultType::TimeOffsetMining => pledge,
-    }
+/// DealWeight and VerifiedDealWeight are spacetime occupied by regular deals and verified deals in a sector.
+/// Sum of DealWeight and VerifiedDealWeight should be less than or equal to total SpaceTime of a sector.
+/// Sectors full of VerifiedDeals will have a SectorQuality of VerifiedDealWeightMultiplier/BaseMultiplier.
+/// Sectors full of Deals will have a SectorQuality of DealWeightMultiplier/BaseMultiplier.
+/// Sectors with neither will have a SectorQuality of BaseMultiplier/BaseMultiplier.
+/// SectorQuality of a sector is a weighted average of multipliers based on their propotions.
+fn sector_quality_from_weight(weight: &SectorStorageWeightDesc) -> SectorQuality {
+    let sector_space_time = weight.sector_size as u64 * weight.duration;
```

Comment:
should probably wrap the sector size in a bigint before doing the multiplication, reason being is you don't want to overflow the u64 on the multiplication